spring-security-oauth2-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/security/oauth2/resource/ResourceServerTokenServicesConfigurationTests.java

```diff
@@ -275,6 +285,41 @@ public void jwtTokenStoreShouldBeConditionalOnMissingBean() throws Exception {
 		assertThat(this.context.getBeansOfType(JwtTokenStore.class)).hasSize(1);
 	}
 
+	@Test
+	public void autoconfigureWhenCustomUserDetailsServiceExposedAsBeanThenJwkTokenStorePicksItUp() throws Exception {
```

Comment:
@jgrandja, what are your thoughts on these tests?

I'm wanting to keep these tests in the spirit of the existing tests in the class (using `SpringApplicationBuilder`) and also avoid standing up a mock web server for the jwk test. Would prefer to simply test Spring Boot's wiring since we likely already have tests for the underlying functionality in `spring-security-oauth`.

Reply:
Agreed. We only need to test that `UserDetailsService` is being picked up and ultimately called. The tests you have is sufficient.